src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java

```diff
@@ -44,36 +44,34 @@ public ValidationService(
      */
     public boolean validatePassword(final String password1, final String password2, Map<String, Object> model) {
 
-        if (StringUtils.isEmpty(password1) && StringUtils.isEmpty(password2)) {
-            ErrorHelper.showError("Error", "public.common.error.password.not.empty", "public.common.error.enter.password", "public.common.error.enter.password", model);
-            return false;
-        }
+        String errorForPassword1 = "";
+        String errorForPassword2 = "";
+
+        // check password 1
         if (StringUtils.isEmpty(password1)) {
-            ErrorHelper.showError("Error", "public.common.error.password.not.empty", "public.common.error.enter.password", "", model);
-            model.put("password2", password2);
-            return false;
-        }
-        if (StringUtils.isEmpty(password2)) {
-            ErrorHelper.showError("Error", "public.common.error.password.not.empty", "", "public.common.error.enter.password", model);
-            model.put("password1", password1);
-            return false;
+            errorForPassword1 = "public.common.error.enter.password";
+        } else if (password1.length() < passwordMinLength || password1.length() > passwordMaxLength || containsIllegalCharacters(password1)) {
```

Comment:
3 different tests here
pass1 too short
pass1 too long
pass1 illegal

Reply:
already in the tests